src/main/java/io/anserini/collection/WikipediaParagraphsCollection.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Anserini: An information retrieval toolkit built on Lucene
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.collection;
+
+import io.anserini.document.WikipediaParagraph;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.wikiclean.WikiClean;
+import org.wikiclean.WikiClean.WikiLanguage;
+import org.wikiclean.WikiCleanBuilder;
+import org.wikiclean.WikipediaBz2DumpInputStream;
+
+/**
+ * Class representing an instance of a Wikipedia paragraphs collection. Note that Wikipedia dumps often come
+ * as a single bz2 file. Since a collection is assumed to be in a directory, place the bz2 file in
+ * a directory prior to indexing.
+ */
+public class WikipediaParagraphsCollection extends Collection<WikipediaParagraph> {
+
+  public class FileSegment extends Collection<WikipediaParagraph>.FileSegment {
+    private final WikipediaBz2DumpInputStream stream;
+    private final WikiClean cleaner;
+    private final Pattern CONTAINS_PERIOD;
+    private final Pattern SINGLE_WORD_WITH_PERIOD;
+    private ArrayList<String> bufferedParagraphs = null;
+    private int currentParagraphIdx = 0;
+    private String title = "";
+
+    protected FileSegment(Path path) throws IOException {
+      this.path = path;
+
+      stream = new WikipediaBz2DumpInputStream(path.toString());
+      cleaner = new WikiCleanBuilder()
+          .withLanguage(WikiLanguage.EN).withTitle(false)
+          .withFooter(false).build();
+
+      CONTAINS_PERIOD = Pattern.compile("\\.");
+      SINGLE_WORD_WITH_PERIOD = Pattern.compile("\\w+\\.");
+    }
+
+    @Override
+    public WikipediaParagraph next() {
+      if (bufferedParagraphs != null && currentParagraphIdx < bufferedParagraphs.size()) {
```

Comment:
I think this block can be put after the `bufferedParagraphs` is populated below?